File: counting_valleys/cv.js
const countigValleys = (steps, path) => {
    let valleys = 0, seaLevel = 0;
    for (const step of path) {
        (step === 'U') ? seaLevel++ : seaLevel--;
        if(seaLevel===0 && step ==='U') valleys++;
    }
    return valleys;
}

console.log(countigValleys(8, 'UDDDUDUU'));
console.log(countigValleys(12, 'DDUUDDUDUUUD'));

File: counting_valleys/cv.py
def countig_valleys(steps: int, path: str) -> int:
    """
    Counting Valleys
    """
    valleys = 0
    level = 0
    for step in path:
        if step == 'U':
            level += 1
        elif step == 'D':
            level -= 1
        if level == 0 and step == 'U':
            valleys += 1
    return valleys

if __name__ == '__main__':
    print(countig_valleys(8, 'UDDDUDUU'))
    print(countig_valleys(12, 'DDUUDDUDUUUD'))